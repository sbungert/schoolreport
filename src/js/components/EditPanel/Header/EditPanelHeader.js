// @flow

import * as React from 'react';
import './EditPanelHeader.css';

export type Props = {
  alert: boolean,
  children?: React.Node,
  subtitle: string,
  count: number,
  title: string,
};

/**
 * Edit Panel header.
 */
export class EditPanelHeader extends React.Component<Props> {
  static defaultProps = {
    alert: false,
    count: -1,
    children: null,
    subtitle: '',
    title: '',
  };

  props: Props;

  render() {
    return (
      <header className="EditPanelHeader">
        <h1 className="EditPanelHeader__headline">
          {this.props.title}
          {this.props.count > -1 && <span data-count={this.props.count}>{this.props.count}</span>}
          <span data-alert={this.props.alert}>{this.props.subtitle}</span>
        </h1>
        {this.props.children && <div className="EditPanelHeader__controls">{this.props.children}</div>}
      </header>
    );
  }
}

export default EditPanelHeader;
