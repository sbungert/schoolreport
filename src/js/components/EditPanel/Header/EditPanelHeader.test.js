// @flow

import React from 'react';
import { shallow } from 'enzyme';
import EditPanelHeader from './EditPanelHeader';

describe('<EditPanelHeader />', () => {
  const props = { title: 'Test title' };

  test('Renders without crashing', () => {
    const wrapper = shallow(<EditPanelHeader {...props} />);
    expect(wrapper).toHaveLength(1);
  });
});
