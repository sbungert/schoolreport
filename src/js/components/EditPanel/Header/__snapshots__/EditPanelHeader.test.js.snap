// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPanelHeader /> Props render correctly 1`] = `
<header
  className="EditPanelHeader"
>
  <h1
    className="EditPanelHeader__headline"
  >
    Test title
    <span
      data-alert={false}
    >
      
    </span>
  </h1>
</header>
`;
